trin-core/src/portalnet/find/iterators/query.rs

```diff
@@ -0,0 +1,156 @@
+use std::time::Instant;
+
+use super::super::query_pool::QueryState;
+
+use discv5::kbucket::Key;
+
+pub trait Query<TNodeId, TResponse, TResult> {
```

Comment:
That's a great callout, because I totally missed that. I need to go back, and account for the functionality provided by `QueryInfo`. Both the `untrusted_enrs` and `callback` fields are also important. I would prefer not to have to add any more methods to the `Query` trait, because it's already sort of my large. I will need to explore some options here.